### DISCUSSION TITLE
Maximum  nodes

### DISCUSSION THREAD

**Discussion by difan0224** [HewlettPackard/swarm-learning#89] [Q&A]
What is maximum number of sl nodes for each sn node?

**Comment by RadhakrishnaJ** [comment]
There is no hard defined limit in software. Following are the two factors which can limit number of SLs.
1. License availability. Once SL count reaches maximum available resources defined on APLS  license management console, further SL containers does not get license to start the execution.
2. If incase licenses are available, then system resources unavailability can limit further SLs.

**Reply by difan0224** [reply]
I notice in the license, the capacity for sl is 16 and sn is 4. Any recommendation set up to fully use these 16 sl nodes? Thanks.
